source/patterns/@aws-solutions-constructs/aws-sns-sqs/lib/index.ts

```diff
@@ -0,0 +1,158 @@
+     */
+    readonly topicProps?: sns.TopicProps,
+    /**
+     * Existing instance of SQS queue object, if this is set then queueProps is ignored.
+     *
+     * @default - Default props are used
+     */
+    readonly existingQueueObj?: sqs.Queue,
+    /**
+     * Optional user provided properties
+     *
+     * @default - Default props are used
+     */
+    readonly queueProps?: sqs.QueueProps,
+    /**
+     * Optional user provided properties for the dead letter queue
+     *
+     * @default - Default props are used
+     */
+    readonly deadLetterQueueProps?: sqs.QueueProps,
+    /**
+     * Whether to deploy a secondary queue to be used as a dead letter queue.
+     *
+     * @default - true.
+     */
+    readonly deployDeadLetterQueue?: boolean,
+    /**
+     * The number of times a message can be unsuccessfully dequeued before being moved to the dead-letter queue.
+     *
+     * @default - required field if deployDeadLetterQueue=true.
+     */
+    readonly maxReceiveCount?: number
+    /**
+     * Use a KMS Key, either managed by this CDK app, or imported. If importing an encryption key, it must be specified in
+     * the encryptionKey property for this construct.
+     *
+     * @default - true (encryption enabled, managed by this CDK app).
+     */
+    readonly enableEncryption?: boolean
+    /**
+     * An optional, imported encryption key to encrypt the SQS queue, and SNS Topic.
+     *
+     * @default - not specified.
+     */
+    readonly encryptionKey?: kms.Key
+}
+
+/**
+ * @summary The SnsToSqs class.
+ */
+export class SnsToSqs extends Construct {
+    public readonly snsTopic: sns.Topic;
+    public readonly encryptionKey?: kms.Key;
+    public readonly sqsQueue: sqs.Queue;
+    public readonly deadLetterQueue?: sqs.DeadLetterQueue;
+
+    /**
+     * @summary Constructs a new instance of the SnsToSqs class.
+     * @param {cdk.App} scope - represents the scope for all the resources.
+     * @param {string} id - this is a a scope-unique id.
+     * @param {SnsToSqsProps} props - user provided props for the construct.
+     * @since 0.8.0
+     * @access public
+     */
+    constructor(scope: Construct, id: string, props: SnsToSqsProps) {
+        super(scope, id);
+
+        // Setup the dead letter queue, if applicable
+        if (props.deployDeadLetterQueue || props.deployDeadLetterQueue === undefined) {
+            const dlq: sqs.Queue = defaults.buildQueue(this, 'deadLetterQueue', {
+                queueProps: props.deadLetterQueueProps
+            });
+            this.deadLetterQueue = defaults.buildDeadLetterQueue({
+                deadLetterQueue: dlq,
+                maxReceiveCount: props.maxReceiveCount
+            });
+        }
+
+        let enableEncryptionParam = props.enableEncryption;
+        let encryptionKeyParam = props.encryptionKey;
+        // Create the encryptionKey if none was provided
+        if (props.enableEncryption !== false) {
```

Comment:
does it require a check for `=== undefined` as well ? If user does not provide this optional parameter